zfsinstaller/zfsinstaller.cpp

```diff
@@ -23,22 +23,201 @@
+		fprintf(stderr, "%s already exists\n", etl_file);
+		fprintf(stderr, "%s has been renamed to %s\n", etl_file, move_etl);
+		return 0;
+	}
+	else {
+		fprintf(stderr, "Error while renaming the file %s\n", etl_file);
+		return 1;
+	}
+}
+
+int ArgParser(int argc, char **argv, long long int &flags, int &levels, int &size_in_mb, std::string &etl_file) {
+	int option_index = 0;
+	while ((option_index = getopt(argc, argv, "l:f:s:p:d")) != -1) {
+		switch (option_index) {
+		case 'p':
+			etl_file = std::string(optarg);
+			break;
+		case 'l':
+			levels = atoi(optarg);
+			break;
+		case 'f':
+			flags = atoll(optarg);
+			break;
+		case 's':
+			size_in_mb = atoi(optarg);
+			break;
+		case 'd':
+			fprintf(stderr, "-d cannot used with other parameters\n");
+			return 1;
+		default:
+			fprintf(stderr, "Incorrect argument provided\n");
+			return 1;
+		}
+	}
+	if (-1 == flags)       flags = 0xffffffff;
+	if (-1 == levels)      levels = 3;
+	if (-1 == size_in_mb)  size_in_mb = 250;
+	if (0 == etl_file.size()) {
+		TCHAR CurrentPath[MAX_PATH_LEN + 1] = L"";
+		DWORD len = GetCurrentDirectory(MAX_PATH_LEN, CurrentPath);
+		int size_needed = WideCharToMultiByte(CP_UTF8, 0, &CurrentPath[0], MAX_PATH_LEN, NULL, 0, NULL, NULL);
+		std::string CwdPath(size_needed, 0);
+		WideCharToMultiByte(CP_UTF8, 0, &CurrentPath[0], MAX_PATH_LEN, &CwdPath[0], size_needed, NULL, NULL);
+		CwdPath.erase(len);
+		etl_file = CwdPath + ETL_FILE;
+	}
+	return 0;
+}
+
+int zfs_log_session_create(int argc, char** argv) {
+	long long int flags = -1;
+	int levels = -1;
+	int size_in_mb = -1;
+	std::string etl_file;
+	char command[MAX_PATH_LEN];
+	int ret;
+
+	ret = ArgParser(argc, argv, flags, levels, size_in_mb, etl_file);
+	if (ret) {
+		fprintf(stderr, "One or more arguments provided is incorrect\n");
+		printUsage();
+		return ret;
+	}
+
+	if (validate_args(flags, levels, size_in_mb, etl_file.c_str())) {
+		fprintf(stderr, "Please check the provided values for the arguments\n");
+		printUsage();
+		return 1;
+	}
+
+	if (0 != session_exists()) { // If Session does not exist
+		if (GetFileAttributesA(etl_file.c_str()) != INVALID_FILE_ATTRIBUTES) { // ETL EXISTS
+			if (0 != move_file(etl_file.c_str())) {
+				fprintf(stderr, "There is an issue creating the session %s\n", LOGGER_SESSION);
+				return 1;
+			}
+		}
+
+		sprintf_s(command, "cmd.exe /c \"logman create trace %s -p {%s} %lld %d -nb 10 10 -bs 10 -mode Circular -max %d -o \"%s\"\" > nul",
+							LOGGER_SESSION, ZFSIN_GUID, flags, levels, size_in_mb, etl_file.c_str());
+
+		ret = system(command);
+		if (ret != 0) {
+			fprintf(stderr, "There is an issue creating the session %s\n", LOGGER_SESSION);
+		}
+		else {
+			fprintf(stderr, "Logman Session %s successfully created\n", LOGGER_SESSION);
+		}
+		return ret;
+	}
+	else {
+		fprintf(stderr, "Logman Session %s already exists\n", LOGGER_SESSION);
+	}
+	return 0;
+}
 
 
 int main(int argc, char* argv[])
 {
-	if (argc <= 2) {
+	if (argc < 2) {

```

Comment:
This check will pass `zfsinstaller install` command, which is not correct (no .inf file).